test/e2e/test/apmserver/checks_apm.go

```diff
@@ -181,47 +220,55 @@ type CountResult struct {
 
 // CheckEventsInElasticsearch checks that the events sent in the previous step have been stored.
 // We only count document to not rely on the internal schema of the APM Server.
-func (c *apmClusterChecks) CheckEventsInElasticsearch(apm apmv1.ApmServer, k *test.K8sClient) test.Step {
-	return test.Step{
-		Name: "Events should eventually show up in Elasticsearch",
-		Test: test.Eventually(func() error {
-			// Fetch the last version of the APM Server
-			var updatedApmServer apmv1.ApmServer
-			if err := k.Client.Get(context.Background(), k8s.ExtractNamespacedName(&apm), &updatedApmServer); err != nil {
-				return err
-			}
+func (c *apmClusterChecks) checkEventsInElasticsearch(apm apmv1.ApmServer, k *test.K8sClient) error {
+	return retry.UntilSuccess(func() error {
```

Comment:
I think that if this is called inside a `test.Eventually(...)` then we may have two nested `UntilSuccess`, which may cancel the first one depending on the timeout values ?

Reply:
Upon looking at this, I've moved the `retry.UntilSuccess` to only be around the assert calls for count, and not around the whole inner function block, which makes more sense.  As for the question about one cancelling the other, I think this is fine, as there's a `test.UntilSuccess` which leverages `*testing.T` and could cancel the whole test, which is what `test.Eventually` uses internally, and there's `retry.UntilSuccess` (very similar names) which is what I'm using on the inner call here, which just returns an error on failure after timeout.

With the recent changes, I can't replicate the test failures you note above using
`make e2e-local TESTS_MATCH=TestAPMServerVersionUpgradeToLatest8x E2E_STACK_VERSION=7.17.2`.  If you continue to replicate these errors, could you let me know how you're calling the e2e tests, so I can fully replicate?  thanks.

Reply:
Ok, I just verified again that these e2e tests are successfully running. Going to kick off in Jenkins.